cmd/experimental/mirror/internal/mirror.go

```diff
@@ -88,39 +103,24 @@ func Mirror(ctx context.Context, src Fetcher, numWorkers uint, store StoreFn) er
 		log.Println("No more work")
 	}()
 
-	go func() {
-		t := time.NewTicker(time.Second)
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			case <-t.C:
-				p := float64(resourcesFetched.Load()*100) / float64(totalResources)
-				log.Printf("Progress %d of %d resources (%0.2f)", resourcesFetched.Load(), totalResources, p)
-			}
-		}
-	}()
-
 	g := errgroup.Group{}
-	for i := range numWorkers {
+	for i := range m.NumWorkers {
 		g.Go(func() error {
 			for j := range work {
 				log.Printf("Worker %d: working on %s", i, j)
 				for ri := range layout.Range(j.from, j.N, srcSize>>(j.level*layout.TileHeight)) {
-					if err := retry.Do(func() error {
-						d, err := src.ReadTile(ctx, j.level, ri.Index, ri.Partial)
-						if err != nil {
+					if err := retry.Do(m.copyTile(ctx, j.level, ri.Index, ri.Partial)); err != nil {
+						log.Println(err.Error())
+						return err
+					}
+					m.resourcesFetched.Add(1)
```

Comment:
`m.resourcesFetched.Add(1)` was called in both `m.copyTile` func and here. Is that expected? The same question applies to `m.copyBundle` below.

Reply:
Nice spot, thanks!